packages/prover/test/unit/utils/conversion.test.ts

```diff
@@ -0,0 +1,90 @@
+import {expect} from "chai";
+import {chunkIntoN} from "../../../src/utils/conversion.js";
+
+describe("utils/conversion", () => {
+  describe("chunkIntoN", () => {
+    const testCases = [
+      {
+        title: "even number of chunks",
+        input: {
+          data: [1, 2, 3, 4, 5, 6],
+          n: 2,
+        },
+        output: [
+          [1, 2],
+          [3, 4],
+          [5, 6],
+        ],
+      },
+      {
+        title: "even number of chunks with additional element",
+        input: {
+          data: [1, 2, 3, 4, 5, 6, 7],
+          n: 2,
+        },
+        output: [[1, 2], [3, 4], [5, 6], [7]],
+      },
+      {
+        title: "odd number of chunks",
+        input: {
+          data: [1, 2, 3, 4, 5, 6],
+          n: 3,
+        },
+        output: [
+          [1, 2, 3],
+          [4, 5, 6],
+        ],
+      },
+      {
+        title: "odd number of chunks with additional element",
+        input: {
+          data: [1, 2, 3, 4, 5, 6, 7],
+          n: 3,
+        },
+        output: [[1, 2, 3], [4, 5, 6], [7]],
+      },
+      {
+        title: "data less than chunk size",
+        input: {
+          data: [1],
+          n: 3,
+        },
+        output: [[1]],
+      },
+      {
+        title: "data 1 less than chunk size",
+        input: {
+          data: [1, 2],
+          n: 3,
+        },
+        output: [[1, 2]],
+      },
+      {
+        title: "data 1 extra than chunk size",
+        input: {
+          data: [1, 2, 3, 4],
+          n: 3,
+        },
+        output: [[1, 2, 3], [4]],
+      },
+      {
+        title: "when data have different order",
```

Comment:
Does this test case make sense? it makes it seem like the type of elements are relevant, rather would add 1-2 tests cases that show that it works with any type of elements, e.g. objects or nested arrays

Reply:
Yes just want to make sure doing chunk does not change the order.

Reply:
All the other existing test cases already assert that. 

My point is that why would the values of elements impact that?

> data have different order

Makes it sounds like this `chunkIntoN` inspects the value of elements, which it does not. IMO better way to prove this is to use different types of elements.